nautobot/extras/jobs.py

```diff
@@ -104,25 +105,185 @@ class Meta:
         """
 
     def __init__(self):
-        self.logger = logging.getLogger(f"{__name__}.{self.__class__.__name__}")
+        self.logger = get_task_logger(self.__module__)
 
-        self.request = None
         self.active_test = "main"
         self.failed = False
         self._job_result = None
 
-        # Compile test methods and initialize results skeleton
-        self.test_methods = []
-
-        for method_name in dir(self):
-            if method_name.startswith("test_") and callable(getattr(self, method_name)):
-                self.test_methods.append(method_name)
+    def __call__(self, *args, **kwargs):
```

Comment:
If we're not even processing the `*args` I'm wondering if perhaps instead we should put `*, **kwargs` here to explicitly disallow positional arguments?
```suggestion
    def __call__(self, *, **kwargs):
```

Reply:
I don't think we'll be able to align on only `kwargs` since celery task chains rely on passing the result of one task as the first positional arg of the next task. That would also be invalid syntax in this context:

```
nautobot-celery_worker-1  |   File "/source/nautobot/extras/jobs.py", line 114
nautobot-celery_worker-1  |     def __call__(self, *, **kwargs):
nautobot-celery_worker-1  |                        ^
nautobot-celery_worker-1  | SyntaxError: named arguments must follow bare *
```